Diagnostics/start-windows-build.ps1

```diff
@@ -0,0 +1,405 @@
+                             -StandardOutput $stdoutFile -StandardError $stderrFile
+        $msg = "Successfully executed: $command"
+    } catch {
+        $msg = "Failed command: $command"
+        $global:PARAMETERS["BUILD_STATUS"] = 'FAIL'
+        $global:PARAMETERS["LOGS_URLS"] += $($stdoutUrl, $stderrUrl)
+        $global:PARAMETERS["FAILED_COMMAND"] = $command
+        Write-Output "Exception: $($_.ToString())"
+        Throw $BuildErrorMessage
+    } finally {
+        Write-Output $msg
+        Write-Output "Stdout log available at: $stdoutUrl"
+        Write-Output "Stderr log available at: $stderrUrl"
+    }
+}
+
+function Get-LatestCommitID {
+    if(!$global:LATEST_COMMIT_ID) {
+        Throw "Failed to get the latest dcos-diagnostics commit ID. Perhaps it has not saved."
+    }
+    return $global:LATEST_COMMIT_ID
+}
+
+function Set-LatestDiagnosticsCommit {
+    Push-Location $DIAGNOSTICS_GIT_REPO_DIR
+    try {
+        if($CommitID) {
+            Start-ExternalCommand { git.exe reset --hard $CommitID } -ErrorMessage "Failed to set Mesos git repo last commit to: $CommitID."
+        }
+        Start-ExternalCommand { git.exe log -n 1 } -ErrorMessage "Failed to get the latest commit message for the Mesos git repo" | Out-File "$DIAGNOSTICS_BUILD_LOGS_DIR\latest-commit.log"
+        $commitId = Start-ExternalCommand { git.exe log --format="%H" -n 1 } -ErrorMessage "Failed to get the latest commit id for the dcos-diagnostics git repo"
+        Set-Variable -Name "LATEST_COMMIT_ID" -Value $commitId -Scope Global  -Option ReadOnly
+    } finally {
+        Pop-Location
+    }
+}
+
+function Clone-DiagnosticsSource {
+    Write-Output "Creating new tests environment"
+    Start-EnvironmentCleanup # Do an environment cleanup to make sure everything is fresh
+    New-Directory $DIAGNOSTICS_DIR
+    New-Directory $DIAGNOSTICS_BUILD_DIR
+    New-Directory $DIAGNOSTICS_BINARIES_DIR
+    New-Directory $DIAGNOSTICS_BUILD_OUT_DIR -RemoveExisting
+    New-Directory $DIAGNOSTICS_BUILD_LOGS_DIR
+    $global:PARAMETERS["BRANCH"] = $Branch
+
+    # Clone DIANOSTICS repository
+    Start-GitClone -Path $DIAGNOSTICS_GIT_REPO_DIR -URL $GitURL -Branch $Branch
+    Set-LatestDiagnosticsCommit
+
+    Start-GitClone -Path $DIAGNOSTICS_DCOS_WINDOWS_GIT_REPO_DIR -URL $DCOS_WINDOWS_GITURL -Branch $Branch
+    Start-GitClone -Path $DIAGNOSTICS_MESOS_JENKINS_GIT_REPO_DIR -URL $MESOS_JENKINS_GIT_URL -Branch "diagnostics"    
+
+    Start-ExternalCommand { git.exe config --global user.email "ostcauto@microsoft.com" } -ErrorMessage "Failed to set git user email"
+    Start-ExternalCommand { git.exe config --global user.name "ostcauto" } -ErrorMessage "Failed to set git user name"
+    #Set-VCVariables "15.0"
+    Write-Output "New tests environment was successfully created"
+}
+
+function Build-DiagnosticsBinaries {
+    Write-Output "Building Diagnostics binary"
+    Push-Location $DIAGNOSTICS_DIR
+    write-output "Current location: $($currLoc.Path)" 
```

Comment:
`$currLoc` is set 3 lines below. I think that `$($currLoc.Path)` will be `$null` here.